src/postgres.rs

```diff
@@ -0,0 +1,179 @@
+use std::{collections::BTreeMap, convert::TryFrom, time::Instant};
+
+use actyxos_sdk::{event::SourceId, Offset, OffsetMap};
+use anyhow::Result;
+use async_trait::async_trait;
+use tokio_postgres::{types::Type, NoTls, Row};
+use tokio_postgres::{Client, Statement};
+use tracing::{debug, info};
+
+use crate::db::{Db, DbConnection, DbEvent, DbEventVec};
+
+pub struct PostgresConnection {
+    client: Client,
+    insert_stmt: Statement,
+    table: String,
+}
+
+#[derive(Debug)]
+pub struct Postgres {
+    host: String,
+    port: u16,
+    user: String,
+    password: String,
+    db_name: String,
+    table: String,
+}
+
+impl Postgres {
+    pub fn new(
+        host: String,
+        port: u16,
+        user: String,
+        password: String,
+        db_name: String,
+        table: String,
+    ) -> Postgres {
+        Postgres {
+            host,
+            port,
+            user,
+            password,
+            db_name,
+            table,
+        }
+    }
+}
+
+#[async_trait]
+impl Db<PostgresConnection> for Postgres {
+    async fn connect(&self) -> Result<PostgresConnection> {
+        info!("Connecting to database");
+        let conn_str = format!(
+            "host={} port={} user={} password={} dbname={}",
+            self.host, self.port, self.user, self.password, self.db_name
+        );
+        let (client, connection) = tokio_postgres::connect(&conn_str, NoTls).await?;
+        tokio::spawn(Box::pin(connection));
```

Comment:
with tokio03 you could get a `JoinHandle` from this, which you could eventually `.abort()`. But this program probably will just exit if there is any issues with the connection, right?

Reply:
yeah, that's why we have `unwrap()` here: https://github.com/Actyx/events-to-db/pull/1/files#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR109